### PR TITLE
feat(variables): link store variables data & refactor dropdown

### PR DESCRIPTION
**Pull request by piggggggggy**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [ ] Not that difficult

### Type of Change
- [x] New feature
- [x] Bug fixes
- [x] Feature improvement
- [x] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] `Error / Warning / Lint / Type`

### Description

SSIA

- Link store data.
- Refactor dropdown, now dropdown accesses store directly without using props from customize page.
- Not that difficult other changes.

### CODE REVIEW

File: src/services/dashboards/dashboard-customize/modules/VariableSelectorDropdown.vue [hunk 1]
```diff
@@ -123,38 +140,27 @@ const handleChangeVisible = () => {
...
-        if (Array.isArray(props.defaultSelected)) {
-            props.defaultSelected.forEach((d) => {
-                defaultSelected.push({ name: d, label: d });
-            });
-        } else defaultSelected.push({ name: props.defaultSelected, label: props.defaultSelected });
-        state.selected = defaultSelected;
+// helper
+const changeVariables = (changedSelected: MenuItem[]) => {
+    const variables = cloneDeep(dashboardDetailState.variables);
+    const reconvertedSelected = changedSelected.map((d) => d.name) as string[];
```
Comment (minseolee) [inline src/services/dashboards/dashboard-customize/modules/VariableSelectorDropdown.vue:154]:
```suggestion
    const reconvertedSelected: string[] = changedSelected.map((d) => d.name);
```
It looks like type assertion is not needed here

**Review by minseolee — APPROVED** [review]
LGTM!

File: src/services/dashboards/dashboard-customize/DashboardCustomizePage.vue [hunk 2]
```diff
@@ -298,13 +294,13 @@ const handleResetVariables = () => {
...
+        if (!originData[d]) return;
         if (isEqual(variableState.variableProperties[d], originSchemaProperties[d])) {
             initialDataWithChangedVariable[d] = originData[d];
         }
     });
-
-    variableState.variableProperties = initialPropertiesWithChangedVariable;
-    variableState.variableData = initialDataWithChangedVariable;
+    dashboardDetailState.variables_schema = { ...dashboardDetailState.variables_schema, properties: initialPropertiesWithChangedVariable };
+    dashboardDetailState.variables = initialDataWithChangedVariable;
```
Comment (WANZARGEN) [inline src/services/dashboards/dashboard-customize/DashboardCustomizePage.vue:303]: suggestion: I think this code is pretty hard to understand..🥲 What about split into two functions for reset flow?

reset order: 
1. reset schema (actually updates only use state)
2. reset variables data

For Example: 
```typescript
const resetSchema = () => {
  const originProperties = variableState.originVariablesSchema.properties;
     // variableState.order is current variable properties' name list
      variableState.order.forEach((property) => {
            updateVariablePropertyUse(property, originProperties[property].use)
      });
}

const updateVariablePropertyUse = (property: string, use: boolean) => {
  dashboardDetailState.variables_schema.properties.[property]?.use = use
}

const resetVariables = () => {
  ...
}

const handleResetVariables = () => {
  resetSchema()
  resetVariables()
}
```

[Scenario]

origin variables: {
  abc: 'abc'
}
origin schema.properties: {
  abc: { use: true }
}

After update:
current schema.properties: {
  abc: { use: false }
}
current variables: {
  abc: 'ddd'
}

After reset:
current schema.properties: {
  abc: { use: true }
}
current variables: {
  abc: 'abc'
}

**Review by WANZARGEN — COMMENTED** [review]
Thanx for your hard work!
I left some suggestion :)

**Comment by piggggggggy** [comment]
Thank you for reviewing illegible code..!
I reflect your reviews.

**Pull request merged**